File: python/java_to_python/07-pgmThread/src/co/edu/campusucc/poo/domain/ThreadExtends.java
package co.edu.campusucc.poo.domain;

public class ThreadExtends extends Thread {
	//Properties
	private int times;

	public void run() {
		System.out.println("Thread Started Running...on UCC University");
		System.out.println("Random Times: " + times);
		printlTimes();
	}

	// Defined Behavior
	public void printlTimes() {
		int timesCount = 0;
		while (timesCount <= times) {
			System.out.print("-");
			timesCount++;
			try {
				Thread.sleep(50);
			} catch (InterruptedException e) {
				e.printStackTrace();
			}
		}
		System.out.println("> OK");
	}

	// Getters and Setters
	public int getTimes() {
		return times;
	}

	public void setTimes(int times) {
		this.times = times;
	}

}

File: python/java_to_python/07-pgmThread/src/co/edu/campusucc/poo/ThreadExtends.py
import threading
import time

class ThreadExtends(threading.Thread):
    def __init__(self):
        threading.Thread.__init__(self)
        self.times = 0

    def run(self):
        print("Thread Started Running...on UCC University")
        print("Random Times: " + str(self.times))
        self.printlTimes()

    # Defined Behavior
    def printlTimes(self):
        timesCount = 0
        while timesCount <= self.times:
            print("-", end="")
            timesCount += 1
            time.sleep(0.05)
        print("> OK")

    # Getters and Setters
    def getTimes(self):
        return self.times

    def setTimes(self, times):
        self.times = times